src/pymodaq/daq_utils/h5modules.py

```diff
@@ -1897,18 +1923,25 @@ def add_datas(self, datas):
                     data_group = self.h5saver.add_data_group(det_group, data_type, metadata=dict(type='scan'))
                 else:
                     data_group = self.h5saver.get_node(det_group, utils.capitalize(data_type))
+
                 for ind_channel, channel in enumerate(datas[data_type]):
                     channel_group = self.h5saver.get_group_by_title(data_group, channel)
                     if channel_group is None:
                         channel_group = self.h5saver.add_CH_group(data_group, title=channel)
+                        # This condition should be added for the 0D case because H5Saver.add_data requires that the key
+                        # data_dict["data"] should be an ndarray, and not a float.
+                        if data_type == "data0D":
+                            datas[data_type][channel]["data"] = np.array(datas[data_type][channel]["data"])
```

Comment:
The last [commits](https://github.com/CEMES-CNRS/PyMoDAQ/pull/79/commits/addec23e326823ba48709a20dd5e26f048e0a561) should solve those issues.
Now the type of the DataToExport["data"] key is checked at the begining of the last for loop and changed to a ndarray of shape (1,) if it is a scalar.
At appending into the EARRAY, it is no more needed to distinguish the "0D" case.

Don’t you think it would be nice to restrict the DataToExport["data"] key to be a ndarray with dimension > 0? (I don’t know if it is usefull somewhere else in the code that it can accept float, int or None). This way the DataToExport objects would for sure be "clean" to be appended within pytables arrays, and one would not have to deal with the particular case of scalar data at this stage.

Reply:
Done in 9d18261f7016bdb4283c19a9800176bd8905b251.